ops/ccs-ops-misc/securityhub/resolve-inactive-findings/resolve-inactive-findings.py

```diff
@@ -0,0 +1,530 @@
+    'security_hub': {
+        'get': 5,
+        'update': 5,
+        'batch_update': 5,
+    },
+    'ec2': {
+        'describe': 25,
+    },
+    's3': {
+        'get': 5,
+    },
+    'rds': {
+        'describe': 25,
+    },
+    'iam': {
+        'list': 5,
+    },
+    'lambda': {
+        'describe': 25,
+    },
+    'kms': {
+        'describe': 25,
+    },
+}
+
+
+# How frequently to update the active resource lists (in minutes)
+UPDATE_INTERVALS = {
+    'AwsEc2Instance': 5,
+    'AwsEc2Volume': 5,
+    'AwsEc2SecurityGroup': 5,
+    'AwsS3Bucket': 5,
+    'AwsRdsDbClusterSnapshot': 60,
+    'AwsRdsDbInstance': 60,
+    'AwsIamUser': 60,
+    'AwsIamPolicy': 60,
+    'AwsIamAccessKey': 5,
+    'AwsAutoScalingAutoScalingGroup': 5,
+    'AwsLambdaFunction': 5,
+    'AwsSnsTopic': 5,
+    'AwsKmsKey': 5,
+}
+
+EC2_INSTANCE_STATES = ['pending', 'running', 'stopping', 'stopped', 'shutting-down', 'terminated']
+ACTIVE_INSTANCE_STATES = set(EC2_INSTANCE_STATES) - set(['terminated'])
+EC2_VOLUME_STATES = ['creating', 'available' , 'in-use' , 'deleting' , 'deleted' , 'error']
+ACTIVE_VOLUME_STATES = set(EC2_VOLUME_STATES) - set(['deleting', 'deleted', 'error'])
+
+# When we resolve a finding, we add a note to explain why.
+RESOLVED_NOTE = "Finding no longer references active resources."
+RESOLVED_BY = "Script"
+
+# This is the filter we use to find findings that need to be resolved. 
+# The 'ResourceType' gets set via command line args.
+FINDING_FILTERS = {
+    'ResourceType': [],
+    'WorkflowStatus': [
+        {
+            'Value': 'NEW',
+            'Comparison': 'EQUALS'
+        },
+        {
+            'Value': 'NOTIFIED',
+            'Comparison': 'EQUALS'
+        },
+    ],
+    'RecordState': [
+        {
+            'Value': 'ACTIVE',
+            'Comparison': 'EQUALS'
+        },
+    ]
+}
+
+# Resource ID regex patterns
+RESOURCE_ID_RE = {
+    'AwsEc2Instance': r'^i-[a-zA-Z0-9]+$',
+    'AwsS3Bucket': r'^[a-z0-9][a-zA-Z0-9-]{1,61}[a-z0-9]$',
+    'AwsEc2Volume': r'^vol-[a-zA-Z0-9]+$',
+    'AwsEc2SecurityGroup': r'^sg-[a-zA-Z0-9]+$',
+    'AwsRdsDbClusterSnapshot': r'^[a-zA-Z0-9-]+$',
+    'AwsRdsDbInstance': r'^[a-zA-Z0-9-]+$',
+    'AwsIamUser': r'^[a-zA-Z0-9-]+$',
+    'AwsIamPolicy': r'^[a-zA-Z0-9-]+$',
+    'AwsIamAccessKey': r'^AWS::IAM::AccessKey:[A-Z0-9]+$',
+    'AwsAutoScalingAutoScalingGroup': r'^[a-zA-Z0-9-]+$',
+    'AwsLambdaFunction': r'^[a-zA-Z0-9-]+$',
+    'AwsSnsTopic': r'^[a-zA-Z0-9-]+$',
+    'AwsKmsKey': r'^[a-zA-Z0-9-]+$',
+}
+
+# SecurityHub Insights are used to get a count of findings that match our filters. This is much faster than
+# paginating through all the findings as there is no .count() method. Insights are dynamically created if
+# needed.
+INSIGHT_NAME_PREFIX = 'NewFailedFindingsBy'
+GROUP_BY = 'AwsAccountId'
+
+
+# Return a boto3 client based on the resource type or 'hub' for a security hub client
+def get_client(region, resource_type):
+    if resource_type.startswith('AwsEc2'):
```

Comment:
This function ought to return a `boto3.client` of one kind or another, but if it's called with an unsupported resource type, this will return `None`. Not too dangerous given the controls elsewhere, but an `else` that raises an exception paired with appropriate exception handling would make this a little more obvious, similar to what you've done in the conditional for `get_active_resources()`

Reply:
Added to the TODO list thanks!